src/validation.h

```diff
@@ -938,6 +938,19 @@ class ChainstateManager
     [[nodiscard]] bool ActivateSnapshot(
         AutoFile& coins_file, const node::SnapshotMetadata& metadata, bool in_memory);
 
+    //! Return the relevant chainstate for a new block.
+    //!
+    //! Because the use of UTXO snapshots requires the simultaneous maintenance
+    //! of two chainstates, when a new block message arrives we have to decide
+    //! which chain we should attempt to append it to.
+    //!
+    //! If our active chainstate hasn't seen the incoming blockhash, return that.
```

Comment:
I find the explanation in the function body more clear. Maybe say:

```md
// Because we expect to have received the headers for the IBD chain
// contents before receiving blocks, when the (shared) block manager
// doesn't include a header for this block, it must go to
// to the snapshot chain.
```